MonkeyWrench.Web.WebService/WebServices.asmx.cs

```diff
@@ -813,6 +813,63 @@ FROM RevisionWork
 		}
 
 		[WebMethod]
+		public GetAuditResponse GetAuditHistory (WebServiceLogin login, int limit, int offset)
+		{
+			GetAuditResponse response = new GetAuditResponse ();
+
+			using (DB db = new DB ()) {
+				Authenticate (db, login, response, true);
+
+				response.AuditEntries = new List<DBAudit> ();
+
+				using (IDbCommand cmd = db.CreateCommand ()) {
+					cmd.CommandText = @"SELECT
+					id,
+					person_id,
+					person_login,
+					ip,
+					stamp,
+					action
+
+					FROM audit";
+					cmd.CommandText += " ORDER BY id DESC ";
+					if (limit > 0)
+						cmd.CommandText += " LIMIT " + limit.ToString ();
+					if (offset > 0)
+						cmd.CommandText += " OFFSET " + offset.ToString ();
+					cmd.CommandText += ";";
+
+					response.Count = GetAuditRecordCount(db);
```

Comment:
You're doing two trips to the database here, instead you can do it all in one query. The `GetLaneForEdit` method is an example how to do it.